src/lib/es2018.asynciterable.d.ts

```diff
@@ -9,17 +9,17 @@ interface SymbolConstructor {
     readonly asyncIterator: unique symbol;
 }
 
-interface AsyncIterator<T, TReturn = any, TNext = undefined> {
+interface AsyncIterator<T, TReturn = any, TNext = unknown> {
     // NOTE: 'next' is defined using a tuple to ensure we report the correct assignability errors in all places.
     next(...args: [] | [TNext]): Promise<IteratorResult<T, TReturn>>;

```

Comment:
Is it worth taking the opportunity to label the tuple element here (and in AsyncIterator/Generator) to avoid `args_0` appearing in the computed function signature?

Reply:
I just looked into doing this, but it causes the number of types we produce to skyrocket in `unionAndIntersectionInference3.ts`, so it's probably better to leave it as is for now.

![image](https://github.com/microsoft/TypeScript/assets/3902892/fb07b349-4d03-40b3-a783-eb4190552b48)

Reply:
Just to close the loop on this one, #59045 fixed this yes?